modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java

```diff
@@ -839,8 +844,25 @@ public StructuredContentField(
 
 		public String getDDMFormFieldDataType() {
 			try {
-				return _ddmStructure.getFieldDataType(
+				String dataType = _ddmStructure.getFieldDataType(
 					_ddmFormFieldValue.getName());
+
+				String displayDataType = dataType;
+
+				if (Objects.equals(dataType, "document-library")) {
+					displayDataType = "document";
```

Comment:
I don't mind having multiple returns in this case, that way you don't have to do the other variable (that you didn't have to do either way)

Reply:
Good point! Changed